### DISCUSSION TITLE
Tags for searching algorithm in crates.io

### DISCUSSION THREAD

**Discussion by alimkoca** [rust-lang/crates.io#5240] [Feature Requests]
Tags would improve searching quality and results, also tags can be subcategories like: `linux/utils`
and can be directory structured like this in a page for choosing them easily:
```
rust/
  - cargo
  - crates.io
  - utils
web/
  - actix-framework
  - websocket
...
```
And with this feature, it will avoid unwanted results, an example happened in community discord server via bot:
```
- ?crate damn
+ Crate damn not found. Did you mean kifuwarabe_tic_tac_toe?
```

**Comment by carols10cents** [comment]
There are already keywords that function much like tags: https://crates.io/keywords 

And categories that are hierarchical: https://crates.io/categories

Also, I'm not sure exactly how adding what you propose would solve the problem of unwanted results-- it could be that no one uses the keyword "damn", or someone could use the keyword "damn" in a way that isn't what *you* expected, so to you the result would be "unwanted", but that's not solved by having tags.

Right now search will put an exact crate name match first in the results, then it searches the crate description, keywords, and README. The kifuwarabe_tic_tac_toe crate has "damn" in its README quite a few times.
